rekall-core/rekall/plugins/linux/heap_analysis.py

```diff
@@ -2874,22 +2871,34 @@ def search_vmas_for_needle(self, search_string=None, search_regex=None,
                 elif search_string:
                     temp['needle'] = search_string
 
-                elif search_regex:
-                    temp['needle'] = search_regex
-
                 result.append(temp)
 
         return result
 
 
+    def is_address_part_of_heap(self, address):
+        """Returns true if the address points at any memory region belonging
+        to the known heap."""
+
+        for vma in self.heap_vmas:
```

Comment:
Use the utils.RangedCollection() for this - it is O(log n) instead of O(n) here.